### DISCUSSION TITLE
Manuelles Anpassen der Verspätung

### DISCUSSION THREAD

**Discussion by PianoRailways** [topological-state/stskit#38] [Ideas]
Es wäre doch sehr nützlich, wenn man die Verspätung eines Zuges an einer bestimmten Station manuell anpassen könnte. Oder so, wie es jTrainGraph hat, ein Menü mit verschiedenen Anpassungsmöglichkeiten: kreuzt, wird überholt (was auch immer der Unterschied ist...) etc.

**Comment by topological-state** [comment]
eine einfache anpassung ist jetzt (10. juli, 20:38) implementiert: im bildfahrplan auf die linie klicken, dann mit +/- die abfahrtsverspätung am ausgangsort verändern.
abhängigkeiten werde ich auch noch implementieren, habe aber noch keine idee, wie ich die bedienung umsetzen soll. vielleicht zwei züge auswählen und dann eine taste drücken? es sind auch toolbuttons dafür geplant - aber die werden wohl noch eine weile warten müssen.

**Comment by topological-state** [comment]
im bildfahrplan können jetzt abhängigkeiten gesetzt werden: zwei übereinander liegende linien nacheinander anklicken, die erste wird gelb (wartender zug), die zweite hellblau (abzuwartender zug). dann einen toolbutton ankunft oder abfahrt abwarten klicken. der gelbe zug richtet sich dann nach dem blauen, plus einer zusätzlichen verzögerung (default +1, mit +/- verstellbar). die abhängigkeit wird speziell markiert und kann wieder gelöscht werden.

bekannte probleme: 

- der abzuwartende zug muss einen fahrplaneintrag haben. 
- wenn der abwartende zug eine durchfahrt hat, wird die abhängigkeit wieder gelöscht, bevor der zug am ausfahrsignal angehalten wird. 
- die markierung der abhängigkeit kann aus mehreren punkten bestehen.
- nahe beieinander liegende linien können manchmal nicht ausgewählt werden.
- bei verknüpften zügen, kann es einige aktualisierungszyklen dauern, bis die richtige verspätung dargestellt wird.

**Comment by PianoRailways** [comment]
Da der Bildfahrplan in Bahnhofsstellwerken (Bern, Olten, Zürich, Luzern, Basel, Lausanne etc...) kaum Verwendung findet, wäre es praktisch, wenn auch in der Gleisbelegung diese Funktionen aktiviert werden könnten (Auswählen von 2 Zügen) um das Abwarten von Anschlüssen oder anderen Zügen in der Gleisbelegung sichtbar zu machen.
